### PR TITLE
Restarting rq-scheduler reschedules all periodics

### PR DESCRIPTION
**Pull request by rauchy**
When `rq-scheduler` restarts, it deletes all existing scheduled jobs and reschedules them. The result of this is that, for example, tasks that should run daily may run more than once a day.

Instead of deleting all jobs and rescheduling them, the `scheduler` command should respect previously scheduled jobs and continue their original iterations.

### CODE REVIEW

File: redash/schedule.py [hunk 1]
```diff
@@ -11,26 +15,36 @@
...
+
+    return hashlib.sha1(json.dumps(metadata, sort_keys=True).encode()).hexdigest()
 
-def schedule(**kwargs):
+
+def prep(kwargs):
     interval = kwargs['interval']
     if isinstance(interval, timedelta):
-        interval = interval.seconds
+        interval = int(interval.total_seconds())
```
Comment (arikfr) [inline redash/schedule.py]: What's the diff between `total_seconds()` and `seconds`?

File: redash/schedule.py [hunk 2]
```diff
@@ -11,26 +15,36 @@
...
+logger = logging.getLogger(__name__)
+
 rq_scheduler = Scheduler(connection=rq_redis_connection,
                          queue_name="periodic",
                          interval=5)
 
+def job_id(kwargs):
+    metadata = kwargs.copy()
+    if 'func' in metadata:
+        metadata['func'] = metadata['func'].__name__
```
Comment (arikfr) [inline redash/schedule.py]: Can there be a job definition without `func`? Also, isn't the func name good enough as a job id?

File: redash/cli/rq.py [hunk 3]
```diff
@@ -9,14 +9,15 @@
...
+from redash.schedule import rq_scheduler, schedule_periodic_jobs, periodic_job_definitions
 
 manager = AppGroup(help="RQ management commands.")
 
 
 @manager.command()
 def scheduler():
-    schedule_periodic_jobs()
+    jobs = periodic_job_definitions()
+    schedule_periodic_jobs(jobs)
```
Comment (arikfr) [inline redash/cli/rq.py]: Why not call the function that returns job definitions in `schedule_periodic_jobs`?

File: redash/schedule.py (hunk 1 above)
Comment (rauchy) [reply redash/schedule.py]: `timedelta(days=1).seconds == 0`

File: redash/schedule.py (hunk 2 above)
Comment (rauchy) [reply redash/schedule.py]: Not sure what I was thinking with checking for existence in metadata.

if we use the function name as an identifier, then we would have to change the function name when we want to apply other things (for example, the interval or results TTL) in order to reschedule them.

File: redash/cli/rq.py (hunk 3 above)
Comment (rauchy) [reply redash/cli/rq.py]: Purely for testability.

File: redash/schedule.py (hunk 1 above)
Comment (arikfr) [reply redash/schedule.py]: :open_mouth:

File: redash/schedule.py (hunk 2 above)
Comment (arikfr) [reply redash/schedule.py]: How about the job id becomes `metadata['func'].__name__:hash`? This way when poking around and seeing a job id it's easy to tell what job it references?

**Pull request merged**